### PR TITLE
build multiarch bootstrap image

### PR DESCRIPTION
**Pull request by zhlhahaha**
build bootstrap image for amd64 and arm64
New Dockerfile for bootstrap has `ARG ARCH` which is not acquired in the build-bootstrap-image test, so the test would always fail.

### CODE REVIEW

**Comment by zhlhahaha** [comment]
cc: @rmohr 
cc: @fgimenez

**Comment by zhlhahaha** [comment]
> @zhlhahaha do we still need multi-arch bootstrap images? If possible it would be great if we can stay with the cross-build (and 

Yes, once we have multi-arch bootstrap images, we can run unit-tests and kind-e2e tests parallelly on the ARM64 server. 

The prow utility images have already supported multi-arch. https://github.com/kubernetes/test-infra/pull/22362
For the Bootstrap image, there is a PR but no maintainer review it. https://github.com/kubernetes/test-infra/pull/23044

File: images/golang/Dockerfile
Comment (dhiller) [inline images/golang/Dockerfile:1]: Don't we need the `ARCH` here? If we don't we would just create a regular amd64 golang image and could thus replace the `publish_multiarch_image.sh` for golang with the regular `publish_image.sh` no?

**Review by dhiller — COMMENTED** [review]
Looks very good already, besides the IMHO missing ARCH inside the golang Dockerfile.

**Comment by zhlhahaha** [comment]
Hi @dhiller the PR has updated, would you like to take a look?

File: images/golang/Dockerfile
Comment (zhlhahaha) [reply images/golang/Dockerfile]: We need multi-arch golang image, I have update the PR.

**Review by dhiller — APPROVED** [review]

**Review by dhiller — COMMENTED** [review]
/approve

/hold @brianmcarey mind taking another look? If you don't find anything, then let's remove the hold

**Comment by zhlhahaha** [comment]
Hi @dhiller 
I see the two required tests `build-bootstrap-image` and `build-golang-image` have failed. As you can see in this PR, the build command for this two images has changed to following command, but in the tests, it still used old version build command. So the tests failed.
```
./publish_multiarch_image.sh bootstrap quay.io kubevirtci
./publish_multiarch_image.sh -l golang quay.io kubevirtci
```

File: github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
Comment (brianmcarey) [inline github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml:91]: As we are doubling the number of images that we are building - we may need to update the default 2 hours timeout for this job. I tried running this locally and it timed out - I updated the timeout to 3 hours and the job passed. Here is an example of how to configure the timeout:
https://github.com/kubevirt/project-infra/blob/9ce9f79571657feabd0847fce5ea058936698022/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml#L110-L112

Once we have the timeout increased for this and the publish job - I think this looks good.

File: images/publish_multiarch_image.sh
Comment (brianmcarey) [inline images/publish_multiarch_image.sh:108]: It would be cool to try this with a tool like buildah but we can try doing that later. Other kubevirt projects are starting to adopt buildah for multi-arch builds - https://github.com/kubevirt/hostpath-provisioner/pull/115

**Review by brianmcarey — COMMENTED** [review]
/lgtm 

/unhold

**Comment by brianmcarey** [comment]
/override build-bootstrap-image

Local run of job completed successfully.

**Review by xpivarc — COMMENTED** [review]
Looks good. I think we should have just one `publish-image` in the end but this will keep us going.

File: github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml
Comment (brianmcarey) [reply github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml:111]: Good point - we probably don't need this specific local image flag.

File: images/publish_multiarch_image.sh
Comment (brianmcarey) [inline images/publish_multiarch_image.sh:86]: If the base image is remote - docker will pull this image as part of the build automatically - @zhlhahaha do we need this special handling for local base image?

Comment (zhlhahaha) [reply images/publish_multiarch_image.sh:86]: Yes, otherwise, docker would not pull correct CPU arch baseimage.

File: github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml
Comment (zhlhahaha) [reply github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-postsubmits.yaml:111]: We need this flag. For Dockerfile whose baseimage is from remote repository, we need manually pull correct CPU arch base image, otherwise it would use local image even if the local image has different CPU arch with `--platform`.
For these Dockerfile we use local built image, like the `golang` image, we do not need to pull base image.

File: images/publish_multiarch_image.sh
Comment (brianmcarey) [reply images/publish_multiarch_image.sh:86]: Ok thanks @zhlhahaha

**Pull request merged**
